lib/analysis_options.3.0.0.yaml

```diff
@@ -0,0 +1,348 @@
+    list_remove_unrelated_type: error
+    literal_only_boolean_expressions: error
+    no_adjacent_strings_in_list: error
+    no_duplicate_case_values: error
+    no_logic_in_create_state: error
+    prefer_void_to_null: error
+    test_types_in_equals: error
+    throw_in_finally: error
+    unnecessary_statements: error
+    unrelated_type_equality_checks: error
+    unsafe_html: error
+    use_build_context_synchronously: error
+    use_key_in_widget_constructors: error
+    valid_regexps: error
+
+dart_code_metrics:
+  metrics-exclude:
+    - test/**
+  rules:
+    - prefer-match-file-name: false
+    - always-remove-listener
+    - avoid-returning-widgets
+    - avoid-unnecessary-setstate
+    - binary-expression-operand-order
+    - no-equal-then-else
+    - prefer-trailing-comma
+    - avoid-passing-async-when-sync-expected
+    - prefer-moving-to-variable:
+        allowed-duplicated-chains: 2
+    - format-comment:
+        only-doc-comments: false
+        ignored-patterns:
+          - ^ @nodoc.*  # Ignores all the comments that start with '@nodoc' (for example: '// @nodoc').
+    - prefer-async-await
+    - avoid-missing-enum-constant-in-map
+    - prefer-match-file-name:
+        exclude:
+            - test/**
+    - avoid-throw-in-catch-block # Use Error.throwWithStackTrace instead throw SomeException.
+    - avoid-unused-parameters
+    - member-ordering:
+        alphabetize: false # TODO set true after code-dart_code_metrics update
+        # Need to fix both - order for widgets and other classes
+        order:
+          - public-static-const-fields
+          - private-static-const-fields
+          - public-static-final-fields
+          - private-static-final-fields
+          - public-static-fields
+          - private-static-fields
+          - public-final-fields
+          - private-final-fields
+          - public-fields
+          - public-getters-setters
+          - private-fields
+          - private-getters-setters
+          - constructors
+          - named-constructors
+          - factory-constructors
+          - overridden-methods
+          - public-static-methods
+          - public-methods
+          - protected-methods
+          - private-static-methods
+          - private-methods
+        widgets-order:
+          - public-static-const-fields
+          - private-static-const-fields
+          - public-static-final-fields
+          - private-static-final-fields
+          - public-static-fields
+          - private-static-fields
+          - public-final-fields
+          - private-final-fields
+          - public-fields
+          - public-getters-setters
+          - private-fields
+          - private-getters-setters
+          - constructors
+          - named-constructors
+          - factory-constructors
+          - overridden-methods
+          - public-static-methods
+          - public-methods
+          - protected-methods
+          - private-static-methods
+          - private-methods
```

Comment:
nit: Not sure if we would like to use exactly the same set of rules for both Widgets and common classes?
I suppose in previous version it was applied implicitly to Widgets. So i guess it's ok.

Reply:
yeah, you're right. Duplication doesn't look good. At first I tried to describe a separate set of rules for `widget-order`, because with the update, the default order changed, and I wanted to bring it to the way it worked before implicitly, so that after the update, all widgets would not be highlighted in blue infos for users of the package.
At first I tried the key fields and methods, but line by line I came to the conclusion that it was necessary to duplicate the rules for widgets from order(